src/main/java/com/amihaiemil/eoyaml/AllYamlLines.java

```diff
@@ -126,23 +140,29 @@ private YamlNode mappingSequenceOrPlainScalar(
             line -> line.trimmed().startsWith("%"),
             line -> line.trimmed().startsWith("!!")
         ).iterator().next();
-        if (first.trimmed().startsWith("-")){
-            node = new ReadYamlSequence(prev, this, guessIndentation);
-        } else if(first.trimmed().contains(":")) {
-            node = new ReadYamlMapping(prev.number(), prev, this,
-                    guessIndentation);
-        } else if(this.original().size() == 1) {
+        Matcher matcher = SEQUENCE_OR_MAP.matcher(first.trimmed());
+        if (matcher.matches()) {
+            // Sequence group 2
+            // Map group 5
+            if (matcher.group(2) != null) {
+                node = new ReadYamlSequence(prev, this, guessIndentation);
+            } else if (matcher.group(5) != null) {
```

Comment:
@newmana Why groups 2 and 5? This should be explained in the JavaDoc above the regex.

Reply:
Done.